extensions/arc/src/ui/dashboards/postgres/postgresParameters.ts

```diff
@@ -496,64 +556,75 @@ export abstract class PostgresParametersPage extends DashboardPage {
 				enabled: false,
 				title: loc.rangeSetting(engineSetting.min!, engineSetting.max!)
 			}).component();
-			valueContainer.addItem(information, { CSSStyles: { 'margin-left': '5px' } });
-		}
-
-		// Can reset individual parameter
-		const resetParameterButton = this.modelView.modelBuilder.button().withProps({
-			iconPath: IconPathHelper.reset,
-			title: loc.resetToDefault,
-			width: '20px',
-			height: '20px',
-			enabled: true
-		}).component();
 
-		this.disposables.push(
-			resetParameterButton.onDidClick(async () => {
-				try {
-					await vscode.window.withProgress(
-						{
-							location: vscode.ProgressLocation.Notification,
-							title: loc.updatingInstance(this._postgresModel.info.name),
-							cancellable: false
-						},
-						async (_progress, _token): Promise<void> => {
-							const session = await this._postgresModel.controllerModel.acquireAzdataSession();
-							try {
-								await this.resetParameter(engineSetting.parameterName!, session);
-							} finally {
-								session.dispose();
-							}
-							await this._postgresModel.refresh();
-						}
-					);
-
-					vscode.window.showInformationMessage(loc.instanceUpdated(this._postgresModel.info.name));
-				} catch (error) {
-					vscode.window.showErrorMessage(loc.instanceUpdateFailed(this._postgresModel.info.name, error));
-				}
-			})
-		);
+			return {
+				parameterName: engineSetting.parameterName!,
+				originalValue: engineSetting.value!,
+				valueComponent: valueComponent,
+				information: information,
+				description: engineSetting.description!,
+				resetButton: resetParameterButton
+			};
+		}
 
-		let parameter: ParametersModel = {
+		return {
 			parameterName: engineSetting.parameterName!,
-			valueContainer: valueContainer,
+			originalValue: engineSetting.value!,
+			valueComponent: valueComponent,
 			description: engineSetting.description!,
 			resetButton: resetParameterButton
 		};
+	}
 
-		return parameter;
+	private collectChangedComponents(name: string): void {
+		if (!this.changedComponentValues.includes(name)) {
+			this.changedComponentValues.push(name);
+		}
 	}
 
-	protected abstract saveParameterEdits(engineSettings: string, session: azdataExt.AzdataSession): Promise<void>;
+	private removeFromChangedComponents(name: string): void {
+		if (this.changedComponentValues.includes(name)) {
+			let index = this.changedComponentValues.indexOf(name);
```

Comment:
You can just do indexOf and check for index !== -1 instead of the includes. That'll save you having to iterate over the array twice